### PR TITLE
Fix problems with claimId in performStepVerification

### LINKED ISSUE #73: in performStepVerification, any claimID can be passed
Might mess things up in the claim manager.

### PR DESCRIPTION
**Pull request by mrsmkl**
The new way of generating the claimId isn't working, yet.
Going to revert some changes next.
Should fix #73

### CODE REVIEW

File: contracts/ClaimManager.sol
Comment (Shrugs) [inline contracts/ClaimManager.sol]: are we sure we want to remove this line?

File: contracts/ClaimManager.sol
Comment (shrugs) [inline contracts/ClaimManager.sol]: we can remove these comments

File: contracts/ClaimManager.sol
Comment (shrugs) [inline contracts/ClaimManager.sol]: nitpick: remove

File: contracts/ClaimManager.sol
Comment (shrugs) [inline contracts/ClaimManager.sol]: @sinahab this comes from the DogeRelay right?

File: contracts/ClaimManager.sol
Comment (shrugs) [inline contracts/ClaimManager.sol]: nitpick: remove

File: contracts/Verifier.sol
Comment (shrugs) [inline contracts/Verifier.sol]: nitpick: remove

File: contracts/Verifier.sol
Comment (shrugs) [inline contracts/Verifier.sol:58]: is there a reason we changed the format of this?

File: contracts/Verifier.sol
Comment (shrugs) [inline contracts/Verifier.sol]: let's make this `camelCase` for linting purposes

File: client/db/config/config.json
Comment (hswick) [inline client/db/config/config.json]: I think this will break on our machines?

File: contracts/ClaimManager.sol
Comment (hswick) [inline contracts/ClaimManager.sol]: I think this could be moved to scryptRunner where it can be computed offchain

Comment (shrugs) [reply contracts/ClaimManager.sol]: do we need to trust that it was computed correctly?

File: contracts/ClaimManager.sol
Comment (hswick) [inline contracts/ClaimManager.sol]: I think we could get away with not hashing the plaintext. that way total parameter size is 96 bytes. As opposed to plaintext being dynamically sized

File: contracts/ClaimManager.sol
Comment (hswick) [reply contracts/ClaimManager.sol]: Well users can pass in the id's anyways. Doesnt matter if its done in scryptVerifier.calcId or in scryptRunner. Better yet we could use web3.utils.sha3

File: contracts/Verifier.sol
Comment (mrsmkl) [reply contracts/Verifier.sol:58]: I tried to work around stack limitation, but it didn't work out.

File: contracts/ClaimManager.sol
Comment (mrsmkl) [reply contracts/ClaimManager.sol]: Good idea.

File: contracts/ClaimManager.sol
Comment (hswick) [inline contracts/ClaimManager.sol]: why is there a bytes at beginning of method?

**Comment by sinahab** [comment]
Nice. This looks great!

Either before or after the merge, we should write a test that demonstrates parallel claims being processed. Don't think we need to hold up the merge on this though, we can do it after.

File: contracts/Verifier.sol
Comment (shrugs) [reply contracts/Verifier.sol:58]: makes sense, thanks

**Review by shrugs — APPROVED** [review]
lgtm, agreed on tests

**Pull request merged**
